pkg/models/roletype/role_type.go

```diff
@@ -9,22 +9,26 @@ import (
 type RoleType string
 
 const (
+	RoleNone   RoleType = "None"
 	RoleViewer RoleType = "Viewer"
 	RoleEditor RoleType = "Editor"
 	RoleAdmin  RoleType = "Admin"
 )
 
 func (r RoleType) IsValid() bool {
-	return r == RoleViewer || r == RoleAdmin || r == RoleEditor
+	return r == RoleViewer || r == RoleAdmin || r == RoleEditor || r == RoleNone
 }
 
 func (r RoleType) Includes(other RoleType) bool {
```

Comment:
Nit: I think we can simplify this check by using a precedence table for role types:

```go
var rolePrecedence = map[RoleType]int{
	None:   1,
	Viewer: 2,
	Editor: 3,
	Admin:  4,
}

func (r RoleType) Includes(other RoleType) bool {
	return rolePrecedence[r] >= rolePrecedence[other]
}
```

We could also use this table to simplify the other functions.

```go
func (r RoleType) IsValid() bool {
	_, ok := rolePrecedence[r]
	return ok
}

// Needed to keep stable order
var roleOrder = [...]RoleType{
	None,
	Viewer,
	Editor,
	Admin,
}

func (r RoleType) Children() []RoleType {
	children := make([]RoleType, 0, 3)

	for _, role := range roleOrder {
		if rolePrecedence[r] > rolePrecedence[role] {
			children = append(children, role)
		}
	}

	return children
}

func (r RoleType) Parents() []RoleType {
	parents := make([]RoleType, 0, 3)

	for _, role := range roleOrder {
		if rolePrecedence[r] < rolePrecedence[role] {
			parents = append(parents, role)
		}
	}

	return parents
}

```